### PR TITLE
Added pass/fail status of job and build

### LINKED ISSUE #404: Include pass/fail/flaky cluster metrics comparison in telemetry notebook
As a data scientist working for AI4CI, I observed that there is a need to utilize the telemetry information extracted for the given build logs. The amount of flaky/passes/failed test can be compared with the telemetry metrics. For example, for the given flaky tests, what is the average memory_usage as compared to the passed/failed tests? 

**Acceptance criteria**
- [ ] Create a jupyter notebbok which shows the comparison of different telemetry metrics with respect to the type of test (as provided from the test_grid data)

### PR DESCRIPTION
**Pull request by suppathak**
## Related Issues and Dependencies

#404 

## This introduces a breaking change

- [ ] Yes
- [x] No

Closes #404

### CODE REVIEW

File: notebooks/data-sources/Telemetry/telemetry_EDA.ipynb [hunk 1]
```diff
@@ -15,11 +15,11 @@
   },
```
Comment (oindrillac) [inline notebooks/data-sources/Telemetry/telemetry_EDA.ipynb:15]: <p>We are able to get the status of past builds from the historical testgrid data that we're collecting. That is how we are calculating build pass/fail metrics over time. </p><p>However, I am not sure if we are able to get the build ID from the Testgrid data. So getting it from the prow logs makes sense. Maybe we should modify the markdown accordingly?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aicoe-aiops/ocp-ci-analysis/pull/470/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

**Review by oindrillac — CHANGES_REQUESTED** [review]
@suppathak I see that the acceptance criteria in #404 also mentions a comparison of different telemetry metrics wrt the type of test.

In this PR, we did extract the type of test, however we still want to compare metrics like CPU usage for different statuses right? Did you plan to include that in a following PR?

File: notebooks/data-sources/Telemetry/telemetry_EDA.ipynb (hunk 1 above)
Comment (suppathak) [reply notebooks/data-sources/Telemetry/telemetry_EDA.ipynb]: Thanks for the feedback @oindrillac :) . I have made changes as per your comment.

**Comment by suppathak** [comment]
> In this PR, we did extract the type of test, however we still want to compare metrics like CPU usage for different statuses right? Did you plan to include that in a following PR?

@oindrillac , I have now included a comparison part for two different job name and build. Thank you. :)

**Comment by chauhankaranraj** [comment]
<p>Building on top of this comment - maybe update the markdown as</p><p><code>... how to get the telemetry data associated with the cluster that ran it.</code></p><p><code>In addition, we will also show how to get the pass/fail status of a given build of a job, at a given point in time. Then, we will compare the telemetry metrics of a passing build with those of a failing build, and see if we can get any insights into the build failure.</code> </p>

---

 <a href='https://app.reviewnb.com/aicoe-aiops/ocp-ci-analysis/pull/470/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by chauhankaranraj** [comment]
<p>Just wondering, would it be more interesting to compare two builds within the same job? So that we can say something like "build 123 of this job failed, but build 456 of the same job passed. here's how the memory usage patterns of these two compare"</p>

---

 <a href='https://app.reviewnb.com/aicoe-aiops/ocp-ci-analysis/pull/470/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by suppathak** [comment]
> What if we plot the metrics for successful and failed builds on the same graph? Maybe it could make it the differences more "obvious" and easy to compare? wdyt?

@chauhankaranraj , Sure I will add that. Good idea! :)

**Comment by chauhankaranraj** [comment]
> Do you think it would be better if I worked on different build , 1 job part in a different notebook/PR? or just add the segment in this notebook itself?

Hmm I'd say if it's cumbersome to find a job where one build is passing and another is failing, then it might be better to defer it to another notebook/PR. But if it's easily available and it's just a matter of changing build id and timestamp, then maybe that could be done in this notebook itself?

**Comment by suppathak** [comment]
> I'm just wondering, would it be okay to display only maybe 4-5 of these dataframes instead of 70 (and then 70 more for the failure cluster)? So that the notebook is a bit more consumable :) wdyt?

Sure it can be done. Thanks

**Review by oindrillac — APPROVED** [review]
thanks for the changes @suppathak 
/lgtm
/approve

/hold for @chauhankaranraj

**Comment by oindrillac** [comment]
black-nb seems to be causing the pre-commit to fail. @harshad16 do you have any tips on how to fix this?

**Comment by harshad16** [comment]
i need some time to investigate it 
will respond about this soon.

**Comment by oindrillac** [comment]
> i need some time to investigate it will respond about this soon.

thanks @harshad16

**Review by chauhankaranraj — APPROVED** [review]
> /hold for @chauhankaranraj

/lgtm
/approve
/unhold

Thanks for the changes @suppathak :smile:

**Comment by oindrillac** [comment]
@suppathak can you reabse this PR on top of #474 so see if the pre-commit issue is fixed?

**Pull request merged**